airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/commands.py

```diff
@@ -108,8 +133,19 @@ async def publish(
     python_registry_token: Secret,
     python_registry_url: str,
     python_registry_check_url: str,
+    promote_release_candidate: bool,
+    rollback_release_candidate: bool,
 ) -> bool:
 
+    if promote_release_candidate and rollback_release_candidate:
+        raise click.UsageError("You can't promote and rollback a release candidate at the same time.")
+    elif promote_release_candidate:
+        rollout_mode = RolloutMode.PROMOTE
+    elif rollback_release_candidate:
+        rollout_mode = RolloutMode.ROLLBACK
+    else:
+        rollout_mode = RolloutMode.PUBLISH
+
```

Comment:
I won't block on this but am still wondering why we're preferring flags for `--promote-release-candidate` and `--rollback-release-candidate` instead of enforcing enums that have `PUBLISH` as the default value.

I know this was @alafanechere 's code originally but if you have an opinion @natikgadzhi I'd be interested!

Reply:
I will take a look!

Reply:
Ok, so, if I understand you correctly, we could do the following: 

1. Implement separate commands for each operation (publish, promote, rollback) — would reuse a lot of code. 
2. Instead of `--promote` and `--rollback`, we could have another option of type enum, i.e. `--operation={publish,promote, rollback}` and default this to publishing. This feels sane! About the same amount of code, but by design can never be both ;) 
3. The design that we have here in this PR already.

I originally thought it's suspicious that you can "promote" or "rollback" a few connectors at the same time, but I do not see a reason to restrict that. 

I think both designs are workable, so I'll continue reading the rest of the PR and prefer pushing forward and verifying this works, I expect we might want a few changes to this in the coming days, so I'll leave it for later.

Reply:
Yes, 2 is what I had in mind. I'm good with punting on the final decision (and they do all seem workable).